### ISSUE TITLE
About training vida source model 

### ISSUE THREAD

**Issue by daeunni** [Yangsenqiao/vida#7]
Hi, @Yangsenqiao, thanks for releasing your code. I really appreciate it. 
Btw, how can I train by myself [this source model](https://drive.google.com/file/d/13R0ZoHaUUXL4PYHWWIwl2NvEcU_3JJxU/view) using your code base? 

Thanks.

**Comment by Yangsenqiao** [comment]
Thank you for your interest in our work. I recommend that you follow these two repositories.
https://github.com/jeonsworld/ViT-pytorch
https://github.com/kentaroy47/vision-transformers-cifar10

**Comment by daeunni** [comment]
Thanks a lot Yangsenqiao! 

Btw, while reimplementing your code and running source_vit.sh with the CIFAR10 dataset, I got pretty different source-only results like below. 

```
[24/03/22 16:27:47] [cifar10c_vit.py:   68]: not resetting model
[24/03/22 16:29:41] [cifar10c_vit.py:   87]: error % [gaussian_noise5]: 88.48%
[24/03/22 16:29:41] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:31:37] [cifar10c_vit.py:   87]: error % [shot_noise5]: 87.51%
[24/03/22 16:31:37] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:33:33] [cifar10c_vit.py:   87]: error % [impulse_noise5]: 92.07%
[24/03/22 16:33:33] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:35:29] [cifar10c_vit.py:   87]: error % [defocus_blur5]: 89.76%
[24/03/22 16:35:29] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:37:26] [cifar10c_vit.py:   87]: error % [glass_blur5]: 89.25%
[24/03/22 16:37:26] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:39:34] [cifar10c_vit.py:   87]: error % [motion_blur5]: 88.75%
[24/03/22 16:39:34] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:41:32] [cifar10c_vit.py:   87]: error % [zoom_blur5]: 89.07%
[24/03/22 16:41:32] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:43:29] [cifar10c_vit.py:   87]: error % [snow5]: 89.86%
[24/03/22 16:43:29] [cifar10c_vit.py:   70]: not resetting model
```

I just utilized these config you provided. 

```
MODEL:
  ADAPTATION: source
  ARCH: Standard_VITB
DATA_DIR: /dataset
TEST:
  BATCH_SIZE: 40
CORRUPTION:
  DATASET: cifar10
  SEVERITY:
    - 5
  TYPE:
    - gaussian_noise
    - shot_noise
    - impulse_noise
    - defocus_blur
    - glass_blur
    - motion_blur
    - zoom_blur
    - snow
    - frost
    - fog
    - brightness
    - contrast
    - elastic_transform
    - pixelate
    - jpeg_compression
```

I would be really appreciate if you have some ideas!

**Comment by Yangsenqiao** [comment]
A little surprising. We will check it.



Best,
Senqiao


---- Replied Message ----
| From | Daeun ***@***.***> |
| Date | 3/22/2024 15:51 |
| To | ***@***.***> |
| Cc | Senqiao Yang ***@***.***> ,
***@***.***> |
| Subject | Re: [Yangsenqiao/vida] About training vida source model (Issue #7) |

Thanks a lot Yangsenqiao!

Btw, while reimplementing your code and running source_vit.sh with the CIFAR10 dataset, I got pretty different source-only results like below.

[24/03/22 16:27:47] [cifar10c_vit.py:   68]: not resetting model
[24/03/22 16:29:41] [cifar10c_vit.py:   87]: error % [gaussian_noise5]: 88.48%
[24/03/22 16:29:41] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:31:37] [cifar10c_vit.py:   87]: error % [shot_noise5]: 87.51%
[24/03/22 16:31:37] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:33:33] [cifar10c_vit.py:   87]: error % [impulse_noise5]: 92.07%
[24/03/22 16:33:33] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:35:29] [cifar10c_vit.py:   87]: error % [defocus_blur5]: 89.76%
[24/03/22 16:35:29] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:37:26] [cifar10c_vit.py:   87]: error % [glass_blur5]: 89.25%
[24/03/22 16:37:26] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:39:34] [cifar10c_vit.py:   87]: error % [motion_blur5]: 88.75%
[24/03/22 16:39:34] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:41:32] [cifar10c_vit.py:   87]: error % [zoom_blur5]: 89.07%
[24/03/22 16:41:32] [cifar10c_vit.py:   70]: not resetting model
[24/03/22 16:43:29] [cifar10c_vit.py:   87]: error % [snow5]: 89.86%
[24/03/22 16:43:29] [cifar10c_vit.py:   70]: not resetting model


I just utilized these config you provided.

MODEL:
  ADAPTATION: source
  ARCH: Standard_VITB
DATA_DIR: /dataset
TEST:
  BATCH_SIZE: 40
CORRUPTION:
  DATASET: cifar10
  SEVERITY:
    - 5
  TYPE:
    - gaussian_noise
    - shot_noise
    - impulse_noise
    - defocus_blur
    - glass_blur
    - motion_blur
    - zoom_blur
    - snow
    - frost
    - fog
    - brightness
    - contrast
    - elastic_transform
    - pixelate
    - jpeg_compression


I would be really appreciate if you have some ideas!

**Comment by daeunni** [comment]
I would be wrong. I will try to see more accurately.

**Comment by Yangsenqiao** [comment]
I just quickly checked the code, and it seems that the checkpoint was not loaded successfully (--checkpoint). I will further check and correct the code.

**Comment by daeunni** [comment]
@Yangsenqiao, friend, I addressed it directly and assigned a checkpoint path here. If some has similar problem, it could be addressed like below. 

```
  base_model = torch.nn.DataParallel(base_model) # make parallel
  checkpoint = torch.load('/d1/daeun/vida/cifar/dataset/vit_base_384_cifar10.t7')
  base_model.load_state_dict(checkpoint['model'], strict=False)
 ```

**Comment by Yangsenqiao** [comment]
Thanks, we will update the code.

**Comment by daeunni** [comment]
Thanks a lot. I think some elements in the config .yaml file are not working.

**Comment by daeunni** [comment]
Senqiao, also may I ask how did you train ViDA source model (WideResNet-28)? (not ViT backbone)

**Comment by Yangsenqiao** [comment]
This repository has the ResNet config, you could follow it.
https://github.com/kentaroy47/vision-transformers-cifar10

**Comment by daeunni** [comment]
Thanks, you are the best!

**Comment by daeunni** [comment]
@Yangsenqiao Hi, friend, could you kindly share your source model training code? 
I assume you attached `ViDAInjectedLinear` to ViT. And my paper, [BECoTTA](https://becotta-ctta.github.io/) shares the same sprits. I would really appreciate if you share your code to reimplement your method clearly. Thanks!

**Comment by Yangsenqiao** [comment]
> @Yangsenqiao Hi, friend, could you kindly share your source model training code? (It is warmup (initialization process) right?) I assume you attached `ViDAInjectedLinear` to ViT. And my paper, [BECoTTA](https://becotta-ctta.github.io/) shares the same sprits. I would really appreciate if you share your code to reimplement your method clearly. Thanks!

Hi, Daeun, I am happy to see your new work :smiley:. Additionally, I have included your paper in our recently organized repository focus on [awesome-CTTA](https://github.com/Yangsenqiao/Awesome-Continual-Test-Time-Adaptation) related paper. I suggest that you cite more related work in your paper to help readers understand the context of this research topic :+1:.

**Comment by Yangsenqiao** [comment]
> Thank you for your interest in our work. I recommend that you follow these two repositories. https://github.com/jeonsworld/ViT-pytorch https://github.com/kentaroy47/vision-transformers-cifar10

And our training code follows these GitHub repositories. We first train the source model, then add the ViDA to the source model, freezing the backbone :snowflake: to tune the adapter :fire:.

**Comment by Yangsenqiao** [comment]
> @Yangsenqiao, friend, I addressed it by directly assigning a checkpoint path here. If someone has similar problem, it could be addressed like below. Thanks for your quick reply.
> 
> ```
>   base_model = torch.nn.DataParallel(base_model) # make parallel
>   checkpoint = torch.load('/d1/daeun/vida/cifar/dataset/vit_base_384_cifar10.t7')
>   base_model.load_state_dict(checkpoint['model'], strict=False)
> ```

These bugs have been fixed. Thanks!

**Comment by TyroneLi** [comment]
> > @Yangsenqiao Hi, friend, could you kindly share your source model training code? (It is warmup (initialization process) right?) I assume you attached `ViDAInjectedLinear` to ViT. And my paper, [BECoTTA](https://becotta-ctta.github.io/) shares the same sprits. I would really appreciate if you share your code to reimplement your method clearly. Thanks!
> 
> Hi, Daeun, I am happy to see your new work 😃. Additionally, I have included your paper in our recently organized repository focus on [awesome-CTTA](https://github.com/Yangsenqiao/Awesome-Continual-Test-Time-Adaptation) related paper. I suggest that you cite more related work in your paper to help readers understand the context of this research topic 👍.

For sure, why have not you released the segmentation part codes yet?????
